src/transport/PairingSession.h

```diff
@@ -50,11 +50,18 @@ class DLL_EXPORT PairingSession
 
     CATValues GetPeerCATs() const { return mPeerCATs; }
 
-    // TODO: the local key id should be allocateed at start
-    // mLocalSessionId should be const and assigned at the construction, such that GetLocalSessionId will always return a valid key
-    // id , and SetLocalSessionId is not necessary.
-    uint16_t GetLocalSessionId() const { return mLocalSessionId; }
-    bool IsValidLocalSessionId() const { return mLocalSessionId != kInvalidKeyId; }
+    Optional<uint16_t> GetLocalSessionId() const
+    {
+        Optional<uint16_t> localSessionId;
+        VerifyOrExit(mSecureSessionHolder, localSessionId = NullOptional);
+        VerifyOrExit(mSecureSessionHolder->GetSessionType() == Transport::Session::SessionType::kSecure,
+                     localSessionId = Optional<uint16_t>::Missing());
+        localSessionId.SetValue(mSecureSessionHolder->AsSecureSession()->GetLocalSessionId());
+    exit:
+        return localSessionId;
+    }
+
+    const SessionHolder & GetSecureSessionHolder() const { return mSecureSessionHolder; }
```

Comment:
I don't think we should expose the internal SessionHolder here. We should return a `SessionHandle` instead.

```
    SessionHandle GetSecureSession() const { return mSecureSessionHolder.Get(); }
```

Reply:
Would I still store mSecureSessionHolder internally though?

Reply:
Ah, I am re-reading your comment.  I think I understand now.

Reply:
@kghost , 

I've pushed up the changes.  Let me know if they are satisfactory.